src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -203,6 +203,28 @@ _get_available_port (void)
   return port;
 }
 
+/**
+ * @brief Internal function to check connection.
+ */
+static bool
+_nns_edge_check_connection (nns_edge_conn_s * conn)
+{
+  GIOCondition condition;
+
+  if (!conn)
+    return false;
```

Comment:
If `GSocket` is NULL or a socket is already closed, then we don't need to check the socket's condition. 
So what about making a code as below?
```suggestion
  if (!conn || !conn->socket || g_socket_is_closed (conn->socket))
    return false;
```

Reply:
Thanks, I updated this condition.

Reply:
Verified. I thanks it makes the code more robust. :100: